.github/workflows/build.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and upload artifacts
+
+on:
+  push:
+    branches:
+      - main
+      - release/**
+  pull_request:
+    branches:
+      - '*'
+
+env:
+  CACHED_BUILD_PATHS: |
+    ${{ github.workspace }}/plugin-build/build/distributions/*.zip
+
+jobs:
+  job_build:
+    name: Build the plugin
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout current commit (${{ github.sha }})
+        uses: actions/checkout@v2
+
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+
+      - name: Check cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CACHED_BUILD_PATHS }}
+          key: ${{ github.sha }}
+
+      - name: Build the plugin
+        run: |
+          cd plugin-build
+          make clean
+          make compile
```

Comment:
I'd do `make all`
and we need to add to make all the `gradlew distZip` step too

Reply:
`make all` also does `dryRelease`, do we want to run it on CI?